### DISCUSSION TITLE
How to debug Codespace secrets being unavailable as env var

### DISCUSSION THREAD

**Discussion by davetapley** [community/community#8978] [Codespaces]
I have [added a Codespace secret](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-encrypted-secrets-for-your-codespaces) and made it available to several repos.

It's available as an env var in one, but not in the other.

I've tried re-creating, but it's still not set.

From trawling the logs (via downloading as `.zip`) I was able to identify these files:

-  `/workspaces/.codespaces/shared/.env`
-  `/workspaces/.codespaces/shared/.user-secrets.json`

and confirmed that they are present and contain the secret on both repos.

Any tips on where else I could look for a clue why it isn't being set on one repo?

---

... it gets weirder:

In the creation logs I also found this snippet:
```sh
if [ -f /workspaces/.codespaces/shared/.env ]; then
   set -o allexport
   . /workspaces/.codespaces/shared/.env
   set +o allexport
fi
```

So I tried to run that manually and I get another error because the `.env` is invalid.
And the `.env` is invalid (because, per [this](https://github.com/codespaces-contrib/codespaces-openvpn#using-the-sample) I'm pasting a multiline `vpnconfig.ovpn` in to the secret). It looks like this:
```
....
GITHUB_USER=davetapley
OPENVPN_CONFIG=client
remote azuregateway..
...
</key>
CODESPACE_NAME=davetapley-...
```

Which raises the questions:
1. Are multiline secrets supposed to be supported?
2. If they aren't (which the design of `/workspaces/.codespaces/shared/.env` seems to imply), then how is it working in one of my repos? 🤯

**Comment by shellerci** [comment]
We ran into this issue around SSH keys, have you tried surrounding your multi-line secrets in single quotes? 
```
GITHUB_USER=davetapley
OPENVPN_CONFIG=client
'remote azuregateway..
...
</key>'
CODESPACE_NAME=davetapley-...
```
